### PR TITLE
Fixes #159: Adds dict support for DefaultFilter

### LINKED ISSUE #159: Bug dict default
CODE:
```
{% set a="1" %}
{% set b=b | default("2") %}
{% set c={"key": "value"} %}
{% set d=d | default({"key": "value"}) %}

Type: {{ type(a) }} - Value: {{ a }}
Type: {{ type(b) }} - Value: {{ b }}
Type: {{ type(c) }} - Value: {{ c }}
Type: {{ type(d) }} - Value: {{ d }}
Type: {{ type(c.key) }} - Value: {{ c.key }}
Type: {{ type(d.key) }} - Value: {{ d.key }}
```
OUTPUT:
```
Type: str - Value: 1
Type: str - Value: 2
Type: dict - Value: {key=value}
Type: str - Value: {key=value}
Type: str - Value: value
Type: null - Value: 
```
What the problem with the last output?

### PR DESCRIPTION
**Pull request by gabru-md**
adds a fix for #159 

prev discussion : #344

### CODE REVIEW

File: src/main/java/com/hubspot/jinjava/lib/filter/DateTimeFormatFilter.java [hunk 1]
```diff
@@ -27,10 +31,19 @@ public String getName() {
 
   @Override
   public Object filter(Object var, JinjavaInterpreter interpreter,
-      String... args) {
+      Object... args) {
+
+    List<String> stringArgs = new ArrayList<>();
```
Comment (jboulter) [inline src/main/java/com/hubspot/jinjava/lib/filter/DateTimeFormatFilter.java]: You could do this all in one loop. In fact, you don't need `stringArgs` at all. I know this was just moved from another file, but might as well clean it up while we're in here.

File: src/main/java/com/hubspot/jinjava/lib/filter/DivideFilter.java [hunk 2]
```diff
@@ -41,11 +42,11 @@
 public class DivideFilter implements Filter {
 
   @Override
-  public Object filter(Object object, JinjavaInterpreter interpreter, String... arg) {
+  public Object filter(Object object, JinjavaInterpreter interpreter, Object... arg) {
     if (arg.length < 1) {
       throw new TemplateSyntaxException(interpreter, getName(), "requires 1 number (divisor) argument");
     }
-    String toMul = arg[0];
+    String toMul = arg[0] == null ? null : Objects.toString(arg[0]);
```
Comment (jboulter) [inline src/main/java/com/hubspot/jinjava/lib/filter/DivideFilter.java]: `Objects.toString` takes care of nulls for you.

File: src/main/java/com/hubspot/jinjava/lib/filter/DefaultFilter.java [hunk 3]
```diff
@@ -61,7 +64,7 @@ public Object filter(Object object, JinjavaInterpreter interpreter, String... ar
       return object;
     }
 
-    return args[0];
+    return (args[0] instanceof PyWrapper) ? args[0] : Objects.toString(args[0]);
```
Comment (jboulter) [inline src/main/java/com/hubspot/jinjava/lib/filter/DefaultFilter.java]: this is the only real significant change, right?

File: src/main/java/com/hubspot/jinjava/lib/filter/DivideFilter.java (hunk 2 above)
Comment (gabru-md) [reply src/main/java/com/hubspot/jinjava/lib/filter/DivideFilter.java]: [Docs](https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html) state that it returns `"null"` and not `null`.
Also if I use `Objects.toString(arg[0])` then the [following test](https://github.com/HubSpot/jinjava/blob/master/src/test/java/com/hubspot/jinjava/lib/filter/AdvancedFilterTest.java#L57) breaks. Therefore had to ensure that `arg[0] == null` check was made.

File: src/main/java/com/hubspot/jinjava/lib/filter/DefaultFilter.java (hunk 3 above)
Comment (gabru-md) [reply src/main/java/com/hubspot/jinjava/lib/filter/DefaultFilter.java]: Yes. basically checking if the argument is an instance of PyWrapper class then we do not need to convert it into String.

File: src/main/java/com/hubspot/jinjava/lib/filter/DateTimeFormatFilter.java (hunk 1 above)
Comment (gabru-md) [reply src/main/java/com/hubspot/jinjava/lib/filter/DateTimeFormatFilter.java]: done :+1:

File: src/main/java/com/hubspot/jinjava/lib/filter/DivideFilter.java (hunk 2 above)
Comment (boulter) [reply src/main/java/com/hubspot/jinjava/lib/filter/DivideFilter.java]: Ah, good to know.

**Pull request merged**